src/test/custom_nodes/node_dynamic_demultiplex.cpp

```diff
@@ -0,0 +1,124 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "../../custom_node_interface.h"
+
+extern "C" {
+
+static const std::string INPUT_TENSOR_NAME = "input_numbers";
+
+int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength) {
+    std::stringstream ss;
+    // validate inputs
+    float* inputTensor = nullptr;
+    size_t valuesPerTensor = 0;
+    for (size_t i = 0; i < inputsLength; ++i) {
+        if (INPUT_TENSOR_NAME == inputs[i].name) {
+            if (inputs[i].dimsLength != 2 ||
+                inputs[i].dims[0] != 1 ||
+                inputs[i].dims[1] == 0) {
+                ss << "improper " << INPUT_TENSOR_NAME
+                   << " dimensions: [" << inputs[i].dims[0] << ", " << inputs[i].dims[1] << "]" << std::endl;
+                std::cout << ss.str() << std::endl;
+                return 1;
+            }
+            ss << "Input valuesPerTensor:" << inputs[i].dims[1] << std::endl;
+            valuesPerTensor = inputs[i].dims[1];
+            inputTensor = reinterpret_cast<float*>(inputs[i].data);
+        } else {
+            std::cout << "Unexpected input" << inputs[i].name << std::endl;
+            return 1;
+        }
+    }
+    if (!inputTensor) {
+        std::cout << "lacking inputs" << std::endl;
+        return 1;
+    }
+    uint32_t demultiplyCount = static_cast<uint32_t>(inputTensor[0]);
```

Comment:
How about using parameter instead of first 4 bytes of an input?

Reply:
We could but you are forced then to change OVMS configuration to have different output size. With current solution you could just pass different inpus to have different output size.